rust/connlib/clients/shared/src/lib.rs

```diff
@@ -267,38 +147,45 @@ where
     }
 }
 
-fn upload_interval() -> Interval {
-    let duration = upload_interval_duration_from_env_or_default();
-    let mut interval = tokio::time::interval_at(Instant::now() + duration, duration);
-    interval.set_missed_tick_behavior(MissedTickBehavior::Skip);
-
-    interval
-}
-
-/// Parses an interval from the _compile-time_ env variable `CONNLIB_LOG_UPLOAD_INTERVAL_SECS`.
+/// Connects to the portal and starts a tunnel.
 ///
-/// If not present or parsing as u64 fails, we fall back to a default interval of 5 minutes.
-fn upload_interval_duration_from_env_or_default() -> Duration {
-    const DEFAULT: Duration = Duration::from_secs(60 * 5);
-
-    let Some(interval) = option_env!("CONNLIB_LOG_UPLOAD_INTERVAL_SECS") else {
-        tracing::warn!(interval = ?DEFAULT, "Env variable `CONNLIB_LOG_UPLOAD_INTERVAL_SECS` was not set during compile-time, falling back to default");
-
-        return DEFAULT;
-    };
-
-    let interval = match interval.parse() {
-        Ok(i) => i,
+/// When this function exits, the tunnel failed unrecoverably and you need to call it again.
+async fn connect<CB>(
+    url: LoginUrl,
+    private_key: StaticSecret,
+    os_version_override: Option<String>,
+    callbacks: CB,
+    max_partition_time: Option<Duration>,
+) where
+    CB: Callbacks + 'static,
+{
+    let tunnel = match Tunnel::new(private_key, callbacks.clone()) {
+        Ok(tunnel) => tunnel,
         Err(e) => {
-            tracing::warn!(interval = ?DEFAULT, "Failed to parse `CONNLIB_LOG_UPLOAD_INTERVAL_SECS` as u64: {e}");
-            return DEFAULT;
+            tracing::error!("Failed to make tunnel: {e}");
+            let _ = callbacks.on_disconnect(&e);
+            return;
         }
     };
 
-    tracing::info!(
-        ?interval,
-        "Using upload interval specified at compile-time via `CONNLIB_LOG_UPLOAD_INTERVAL_SECS`"
+    let portal = PhoenixChannel::connect(
+        Secret::new(url),
+        get_user_agent(os_version_override),
+        PHOENIX_TOPIC,
+        (),
+        ExponentialBackoffBuilder::default()
+            .with_max_elapsed_time(max_partition_time)
+            .with_max_interval(MAX_RECONNECT_INTERVAL)
+            .build(),
     );
 
-    Duration::from_secs(interval)
+    let mut eventloop = Eventloop::new(tunnel, portal);
+
+    match std::future::poll_fn(|cx| eventloop.poll(cx)).await {
+        Ok(never) => match never {},
```

Comment:
can this be changed to:
```suggestion
        Ok(_) => {},
```

Reply:
It can but then we are not asserting that `Ok` is never actually hit and a refactoring of the return value would silently drop it!